exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java

```diff
@@ -0,0 +1,98 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.exporters.jaeger;
+
+import io.grpc.ManagedChannel;
+import io.opentelemetry.exporters.jaeger.proto.api_v2.Collector;
+import io.opentelemetry.exporters.jaeger.proto.api_v2.CollectorServiceGrpc;
+import io.opentelemetry.exporters.jaeger.proto.api_v2.Model;
+import io.opentelemetry.proto.trace.v1.Span;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+
+/** Exports spans to Jaeger via gRPC, using Jaeger's protobuf model. */
+public class JaegerGrpcSpanExporter implements SpanExporter {
+  private final CollectorServiceGrpc.CollectorServiceBlockingStub blockingStub;
+  private final Model.Process process;
+
+  /**
+   * Creates a new Jaeger gRPC Span Reporter with the given name, using the given channel.
+   *
+   * @param serviceName this service's name
+   * @param channel the channel to use when communicating with the Jaeger Collector
+   */
+  public JaegerGrpcSpanExporter(String serviceName, ManagedChannel channel) {
+    String hostname;
+    String ipv4;
+
+    if (serviceName == null || serviceName.trim().length() == 0) {
+      throw new IllegalArgumentException("Service name must not be null or empty");
+    }
+
+    try {
+      hostname = InetAddress.getLocalHost().getHostName();
+      ipv4 = InetAddress.getLocalHost().getHostAddress();
+    } catch (UnknownHostException e) {
+      hostname = "(unknown)";
+      ipv4 = "0.0.0.0";
+    }
+
+    Model.KeyValue clientTag =
+        Model.KeyValue.newBuilder().setKey("jaeger.version").setVStr("opentelemetry-java").build();
+
+    Model.KeyValue ipv4Tag = Model.KeyValue.newBuilder().setKey("ip").setVStr(ipv4).build();
+
+    Model.KeyValue hostnameTag =
+        Model.KeyValue.newBuilder().setKey("hostname").setVStr(hostname).build();
+
+    this.process =
+        Model.Process.newBuilder()
+            .setServiceName(serviceName)
+            .addTags(clientTag)
+            .addTags(ipv4Tag)
+            .addTags(hostnameTag)
+            .build();
+    this.blockingStub = CollectorServiceGrpc.newBlockingStub(channel);
+  }
+
+  @Override
+  public ResultCode export(List<Span> spans) {
+    Model.Batch.Builder builder = Model.Batch.newBuilder();
+    builder.addAllSpans(Adapter.toJaeger(spans));
+    builder.setProcess(this.process);
+
+    Collector.PostSpansRequest.Builder requestBuilder = Collector.PostSpansRequest.newBuilder();
+    requestBuilder.setBatch(builder.build());
+    Collector.PostSpansRequest request = requestBuilder.build();
+
+    try {
+      // for now, there's nothing to check in the response object
+      //noinspection ResultOfMethodCallIgnored
+      this.blockingStub.postSpans(request);
+      return ResultCode.SUCCESS;
+    } catch (Throwable t) {
```

Comment:
You need to catch `io.grpc.StatusRuntimeException` and extract the `io.grpc.Status` from it. Based on the status we can decide which one are retriable vs not-retriable.

Reply:
I'll check that, but I don't think the Jaeger Collector ever fails for this call: all it does is to enqueue the spans for async processing, returning an empty response to the client.

This means that only client/networking issues might occur here, and those would be handled by the gRPC client already, as I understand.

Reply:
I just double checked the collector's code, and I was mostly right: in the default case, Jaeger won't fail. When a specific option is set (`reportBusy`), Jaeger might fail with an unknown status in case the work queue is full. In that case, I think the client should not retry the request, as the server is clearly already busy.

cc @yurishkuro: are you using that option internally? when you get such error, what do you currently do? Perhaps you have some sort of exponential back-off?

Reply:
Sounds good. I added the two concrete cases you mentioned as retryable (UNAVAILABLE, DEADLINE_EXCEEDED).